modules/light-clients/08-wasm/types/client_state.go

```diff
@@ -100,21 +102,23 @@ func (cs ClientState) GetTimestampAtHeight(
 // Initialize checks that the initial consensus state is an 08-wasm consensus state and
 // sets the client state, consensus state in the provided client store.
 // It also initializes the wasm contract for the client.
-func (cs ClientState) Initialize(ctx sdk.Context, _ codec.BinaryCodec, clientStore sdk.KVStore, state exported.ConsensusState) error {
+func (cs ClientState) Initialize(ctx sdk.Context, cdc codec.BinaryCodec, clientStore sdk.KVStore, state exported.ConsensusState) error {
 	consensusState, ok := state.(*ConsensusState)
 	if !ok {
 		return errorsmod.Wrapf(clienttypes.ErrInvalidConsensus, "invalid initial consensus state. expected type: %T, got: %T",
 			&ConsensusState{}, state)
 	}
 
+	// Do not allow initialization of a client with a code hash that hasn't been previously stored via storeWasmCode.
+	if !HasCodeHash(ctx, cdc, cs.CodeHash) {
```

Comment:
note for future: have test coverage of this line

Reply:
has the future arrived? Should we open an issue?

Reply:
hello from the future #4838